src/transaction/log_applier.c

```diff
@@ -7043,141 +7043,175 @@ la_print_log_arv_header (const char *database_name, LOG_ARV_HEADER * hdr, bool v
+ *   return: NO_ERROR or error code
+ *   database_name(in): db name
+ *   log_path(in): real path of log file
+ *   page_num(in): the number of log page 
+ *   verbose(in): applyinfo -v option
+ *   copied_eof_lsa(in/out): inforamtion for delayed log count
+ *   copied_append_lsa(in/out): current append location
+ */
+int
+la_get_copied_log_info (const char *database_name, const char *log_path, INT64 page_num, bool verbose,
+			LOG_LSA * copied_eof_lsa, LOG_LSA * copied_append_lsa)
+{
+  int error = NO_ERROR;
+  char active_log_path[PATH_MAX];
+  char log_path_buf[PATH_MAX];
+
+  assert (database_name != NULL);
+  assert (log_path != NULL);
+
+  if (realpath (log_path, log_path_buf) != NULL)
     {
-      printf ("%s\n", db_error_string (3));
+      log_path = log_path_buf;
+    }
+
+  memset (active_log_path, 0, PATH_MAX);
+  /* init la_Info */
+  la_init (log_path, 0);
+
+  printf ("\n *** Copied Active Info. *** \n");
+
+  fileio_make_log_active_name ((char *) active_log_path, la_Info.log_path, database_name);
+  if (!fileio_is_volume_exist ((const char *) active_log_path))
+    {
+      er_set_with_oserror (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_LOG_MOUNT_FAIL, 1, active_log_path);
+      error = ER_LOG_MOUNT_FAIL;
+      goto check_copied_log_volume_info_end;
     }
-  error = NO_ERROR;
 
-  if (check_copied_info)
+  /* read copied active page */
+  error = la_find_log_pagesize (&la_Info.act_log, la_Info.log_path, database_name, false);
+  if (error != NO_ERROR)
     {
-      /* check active log file */
-      memset (active_log_path, 0, PATH_MAX);
-      fileio_make_log_active_name ((char *) active_log_path, la_Info.log_path, database_name);
-      if (!fileio_is_volume_exist ((const char *) active_log_path))
-	{
-	  er_set_with_oserror (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_LOG_MOUNT_FAIL, 1, active_log_path);
-	  error = ER_LOG_MOUNT_FAIL;
-	  goto check_copied_info_end;
-	}
+      goto check_copied_log_volume_info_end;
+    }
 
-      /* read copied active page */
-      error = la_find_log_pagesize (&la_Info.act_log, la_Info.log_path, database_name, false);
-      if (error != NO_ERROR)
-	{
-	  goto check_copied_info_end;
-	}
+  *copied_eof_lsa = la_Info.act_log.log_hdr->eof_lsa;
+  *copied_append_lsa = la_Info.act_log.log_hdr->append_lsa;
 
-      *copied_eof_lsa = la_Info.act_log.log_hdr->eof_lsa;
-      *copied_append_lsa = la_Info.act_log.log_hdr->append_lsa;
+  la_print_log_header (database_name, la_Info.act_log.log_hdr, verbose);
 
-      printf ("\n *** Copied Active Info. *** \n");
-      la_print_log_header (database_name, la_Info.act_log.log_hdr, verbose);
+check_copied_log_volume_info_end:
+  if (error != NO_ERROR)
+    {
+      printf ("%s\n", db_error_string (3));
+      /*If error occured about the -L option, it is not necessary to check log page information. So, fuction is end here */
```

Comment:
the comments are added for the explanation; -L option returns error and -p option returns NO_ERROR.